src/System.Diagnostics.EventLog/tests/EventLogEntryCollectionTests.cs

```diff
@@ -0,0 +1,124 @@
+                }
+            }
+            finally
+            {
+                EventLog.DeleteEventSource(source);
+                EventLog.Delete(log);
+            }
+        }
+
+        [ConditionalFact(typeof(Helpers), nameof(Helpers.IsElevatedAndSupportsEventLogs))]
+        public void CheckingEntryEqualityWithNull()
+        {
+            string log = "NullTest";
+            string source = "Source_" + nameof(CheckingEntryEqualityWithNull);
+
+            try
+            {
+                EventLog.CreateEventSource(source, log);
+                using (EventLog eventLog = new EventLog())
+                {
+                    eventLog.Source = source;
+                    eventLog.WriteEntry(message);
+                    EventLogEntry entry = eventLog.Entries[eventLog.Entries.Count - 1];
+                    Assert.False(entry.Equals(null));
+                }
+            }
+            finally
+            {
+                EventLog.DeleteEventSource(source);
+                EventLog.Delete(log);
+            }
+        }
+
+        [ConditionalFact(typeof(Helpers), nameof(Helpers.IsElevatedAndSupportsEventLogs))]
+        public void CheckingEntryEqualityAndIndex()
+        {
+            string log = "IndexTest";
+            string source = "Source_" + nameof(CheckingEntryEqualityAndIndex);
+
+            try
+            {
+                EventLog.CreateEventSource(source, log);
+                using (EventLog eventLog = new EventLog())
+                {
+                    eventLog.Source = source;
+                    eventLog.WriteEntry(message);
+                    EventLogEntry entry = eventLog.Entries[eventLog.Entries.Count - 1];
+                    Assert.True(entry.Equals(entry));
+                    eventLog.WriteEntry(message);
+                    EventLogEntry secondEntry = eventLog.Entries[eventLog.Entries.Count - 1];
+                    Assert.Equal(entry.Index + 1, secondEntry.Index);
+                }
+            }
+            finally
+            {
+                EventLog.DeleteEventSource(source);
+                EventLog.Delete(log);
+            }
+        }
+
+        [ConditionalFact(typeof(Helpers), nameof(Helpers.IsElevatedAndSupportsEventLogs))]
+        public void CheckingEntryInEquality()
+        {
+            string log = "InEqualityTest";
+            string source = "Source_" + nameof(CheckingEntryInEquality);
+
+            try
+            {
+                EventLog.CreateEventSource(source, log);
+                using (EventLog eventLog = new EventLog())
+                {
+                    eventLog.Source = source;
+                    eventLog.WriteEntry(message);
+                    eventLog.WriteEntry(message);
+                    EventLogEntry entry = eventLog.Entries[eventLog.Entries.Count - 1];
+                    EventLogEntry secondEntry = eventLog.Entries[eventLog.Entries.Count - 2];
+                    Assert.False(entry.Equals(secondEntry));
```

Comment:
how is tihs testing inequality?

Reply:
It checks whether the eventlogentry instance is same or not..it doesnot check the content

Reply:
Oh, I see it's Assert.False. misread.